python/pyarrow/types.pxi

```diff
@@ -326,6 +326,22 @@ cdef class DataType(_Weakrefable):
         """
         return self.type.layout().buffers.size()
 
+    @property
+    def is_variadic(self):
```

Comment:
should we call this something different than `is_variadic` ? @jorisvandenbossche @pitrou thoughts?

Reply:
`has_variadic_buffers` would sound more explicit